docs/source/cuda/cuda_array_interface.rst

```diff
@@ -69,13 +69,213 @@ The following are optional entries:
+   implies that:
+
+   - The CAI design, and the design and implementation of the Producer and
+     Consumer do not specify or guarantee correctness with respect to data
+     races.
+   - Instead, the User is responsible for ensuring correctness with respect to
+     data races.
+4. All operations and exchange of data are orchestrated from a single host
+   thread. Use of multiple threads on the host is outside of the scope of this
+   version of the CAI. A future revision to the interface specification may
+   consider the use of multiple host threads once the current version and its
+   implementations have matured and become well-understood.
+
+
+Interface Requirements
+~~~~~~~~~~~~~~~~~~~~~~
+
+The ``sync`` and ``stream`` items of the interface indicate whether and how
+synchronization is expected so that producers and consumers that operate
+asynchronously can avoid hazards when exchanging data through the CUDA Array
+Interface.
+
+From the Consumer's perspective:
+
+* ``stream`` indicates the stream on which the Producer may have in-progress
+  operations on the data, and which the Consumer may be expected to either:
+
+  - synchronize on before accessing the data, or,
+  - enqueue operations in when accessing the data.
+
+  Whether one of these requirements applies depends partly on the ``sync``
+  parameter, and partly on the User's choice.
+
+* ``sync`` indicates the requirement for synchronization on the given
+  ``stream``.
+
+  - When ``sync`` is ``False``, or the User has set the Consumer to ignore CAI
+    synchronization semantics, the Consumer may assume that operating
+    on the data is immediately permitted, with no further synchronization.
+  - When ``sync`` is ``True``, and the User has not overridden CAI
+    synchronization semantics, then the Consumer must either:
+
+    - Synchronize on the provided stream prior to accessing the data. If after
+      synchronization the Consumer operates on the data in another stream, it
+      must ensure that no computation can take place in the provided stream
+      until its operations in its own choice of stream have taken place. This
+      could be achieved by either:
+
+      - Placing a wait on an event in the provided stream that occurs once all
+        of the Consumer's operations on the data are completed, or
+      - Avoiding returning control to the user code until after its operations
+        on its own stream have completed.
+
+    - Only enqueuing operations on the data in the provided stream. The Consumer
+      may return control to the User code immediately after enqueueing its work,
+      as the work will all be serialized on the exported array's stream, ensuring
+      correctness even if the User code were to induce the Producer to
+      subsequently start enqueueing more work on the same stream.
+
+When exporting an array through the CAI, Producers must ensure that:
+
+* There is work on the data enqueued on at most one stream.
```

Comment:
This is an implementation detail right?  For instance, the producer could:

1. Enqueue work on streams 7, 9, 15
2. Enqueue events immediately following the work on those streams
3. Enqueue wait-on-event on stream 3; and set **stream** to 3

?

In other words, the end result should be that the buffer is available to on **stream**, but that does not imply that all the work on the buffer actually happens on **stream**.

Reply:
+1

Reply:
You're right, there was no need for the producer to just use one stream. The wording is changed to allow multiple streams, as long as synchronizing on the provided stream is sufficient to sync with all streams, using your example above @sanjoy